policy/policy-authority.go

```diff
@@ -51,6 +62,42 @@ func New(dbMap *gorp.DbMap, enforceWhitelist bool, challengeTypes map[string]boo
 	return &pa, nil
 }
 
+type blacklistJSON struct {
+	Blacklist []string
+}
+
+// SetHostnamePolicyFile will load the given policy file, returning error if it
+// fails. It will also start a reloader in case the file changes.
+func (pa *AuthorityImpl) SetHostnamePolicyFile(f string) error {
+	_, err := reloader.New(f, pa.loadHostnamePolicy)
+	return err
+}
+
+func (pa *AuthorityImpl) loadHostnamePolicy(b []byte, err error) error {
+	if err != nil {
+		pa.log.Err(fmt.Sprintf("loading hostname policy: %s", err))
```

Comment:
Um, this feels fishy. Does reloader give stale data in case of an error? If so, don't you already have the stale data processed, so you can return early? If not, you're replacing stale data with garbage.